stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2495,38 +2482,62 @@ collections = [
 ]
 }%
 
+% for (Self, a_self, TypeParametersDecl, TypeParameters, AnyTypeParameters, Sequence, AnySequenceType) in collections:
+
 /// Header part of the native storage.
-internal struct _HashedContainerStorageHeader {
+internal struct _${Self}StorageHeader<${TypeParameters}> {
```

Comment:
Yeah I went back on forth on this a few times; always happy to delete more types but definitely better for a followup.